src/main.cpp

```diff
@@ -454,15 +472,44 @@ void setup() {
     antenna_init();
 #endif
 
+    // ======================================

```

Comment:
is this code yours, or pasted from somewhere? If pasted, do we have licencse compatible to Apache 2.0 for using it?

Reply:
Don't worry it's my code (just reading I2C device)
For OLED I have a board without OLED and undefined the OLED constant and I wanted to keep flash code as small as possible so I though (I admit it's not cleaner) removing all reference to OLED (when not needed) was a good idea since OLED code can be large. 
would be interesting to see code difference with NULL OLED and with all references removed, just curious

Reply:
Ok, i see.
Can you check how much the difference is in code size without u8x8?

I think too many compiler directives are not a good solution in this. Better way would be probably to capsule all display writes in own function, which calls u8x8 when OLED is present, otherwise does nothing.

Reply:
Could you remove this OLED regarding commit from your pull request? Then i could merge the PR to master. Thank you.

Reply:
With no reference to u8g library
![image](https://user-images.githubusercontent.com/2471931/38255106-40cc8b9a-375b-11e8-8298-96e5e44aab5a.png)

With NULL display
![image](https://user-images.githubusercontent.com/2471931/38255142-52fb9388-375b-11e8-9f9e-c35aaf53f275.png)

Libray is taking flash size (but we have plenty) so no matter for now.

I Removed all HAS_DISPLAY as suggested

Reply:
Yes, but is only ~6Kb of flash, that is suprisingly sparse. So we keep the u8x8 null device. Since the author of u8g2 lib did implement the null device exclusively  on my request in his v2.21.7 :-)
NB: That's why for boards without display i added in platformio.ini

lib_deps = 
    U8g2@>2.21.7

since u8g2 versions prior to 2.21.7 do not have the null device.

Reply:
btw, thanks for contributing again. I will merge your PR now.

Reply:
You're welcome, really nice project, I sure we can improve it again with lot of features ;-)

Reply:
BTW, I used to use u8g library but now I've migrated all my project to this lib
https://github.com/ThingPulse/esp8266-oled-ssd1306

Check out the video on this project (and the small code needed to do it), I really like the "swipe mod"
https://github.com/hallard/Scan-I2C-WiFi